Sources/hostmgr/commands/benchmark/NetworkBenchmark.swift

```diff
@@ -6,41 +6,53 @@ import libhostmgr
 
 private let startDate = Date()
 
-struct NetworkBenchmark: ParsableCommand {
+struct NetworkBenchmark: AsyncParsableCommand {
 
     static let configuration = CommandConfiguration(
         commandName: "network",
         abstract: "Test Network Speed"
     )
 
-    func run() throws {
-        guard let file = try VMRemoteImageManager().list().sorted(by: { $0.size < $1.size }).first else {
+    private static let limiter = Limiter(policy: .throttle, operationsPerSecond: 1)
+
+    func run() async throws {
+        let remoteImages = try await RemoteVMRepository().listImages(sortedBy: .size)
+
+        guard let file = remoteImages.last else {
             throw CleanExit.message("Unable to find a remote image to use as a network benchmark")
         }
 
-        try S3Manager().streamingDownloadFile(
-            region: Configuration.shared.vmImagesRegion,
+        let manager = S3Manager(
             bucket: Configuration.shared.vmImagesBucket,
-            key: file.imageKey,
-            destination: URL(fileURLWithPath: "/dev/null"),
-            progressCallback: self.showProgress
+            region: Configuration.shared.vmImagesRegion
+        )
+
+        try await manager.download(
+            object: file.imageObject,
+            to: FileManager.default.temporaryFilePath(),
+            progressCallback: self.updateProgress
```

Comment:
I'm wondering if it would make the progress reporting code simpler by returning Alamofire's `Progress` instance, instead of our custom `FileTransferProgress`. Having a `Progress` instance would make it possible to using Combine's `debounce` operator, rather than having our own implementation.

```swift
// pseudo code

var progressSubscription: Cancellable? = nil
try await manager.download(
    object: file.imageObject,
    to: FileManager.default.temporaryFilePath(),
    progressCallback: { progress in
        guard progressSubscription = nil else {
            return
        }

        progressSubscription = progress
            .publisher(for: \.fractionCompleted)
            .debounce(for: .seconds(1), scheduler: DispatchQueue.main)
            .sink { _ in
                let downloadedSize = progress.completedUnit
                let totalSize = progress.totalUnit
                let rate = ...
                logger.info("Downloaded \(downloadedSize) of \(totalSize) [Rate: \(rate) per second]")
            }
    }
)
progressSubscription.cancel()
```

Reply:
This would definitely be nice, but unfortunately the Alamofire progress doesn't seem to populate the `throughput` field, which we need for debugging (to be able to see how well the network is performing, in particular). 

If it did, I'd be super happy to drop a whole bunch of code for sure! 

In a forthcoming PR, it should actually be possible to drop the `Limiter` altogether with some better output handling.